rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JIRI.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+/**
+ * Marker interface for RDF4J implementations of IRI.
+ * 
+ */
+public interface RDF4JIRI 
+	extends RDF4JTerm<org.eclipse.rdf4j.model.IRI>, org.apache.commons.rdf.api.IRI {
+	
+	/**
+	 * Return the corresponding RDF4J {@link org.eclipse.rdf4j.model.IRI}.
+	 * 
+	 * @return The corresponding RDF4J IRI
+	 */
+	public org.eclipse.rdf4j.model.IRI asValue();
```

Comment:
I removed the null-overrides and instead clarified the specialized generic parameter in the interface javadoc.